keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -143,4 +158,81 @@ void fetchingNonExistentSecretReturnsEmpty() {
     Optional<String> secret = awsSecretsManagerDefault.fetchSecret("signers-aws-integration/empty");
     assertThat(secret).isEmpty();
   }
+
+  void validateMappedSecret(
+      final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries,
+      final String secretName) {
+    final Optional<AbstractMap.SimpleEntry<String, String>> secretEntry =
+        secretEntries.stream().filter(e -> e.getKey().equals(secretName)).findAny();
+    assertThat(secretEntry).isPresent();
+    assertThat(secretEntry.get().getValue()).isEqualTo(SECRET_VALUE);
+  }
+
+  @Test
+  void multipleSecretsCanBeFetchedAndMapped() {
+    createSecret(); // 2nd
+    createSecret(); // 3rd
+
+    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecretsList(AbstractMap.SimpleEntry::new);
+
+    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
```

Comment:
Can the size also be checked? I'm a bit dubious that this does this actually multiple secrets as the same key is backed by a set in the mapping function.

Reply:
Might be tricky with other unit tests running in parallel. Although more unique secret names per unit test would resolve that.